### PR TITLE
Bump ruff version

### PR DESCRIPTION
**Pull request by danielhollas**
Most of the changes here are minor import sorting changes (I checked all of them and they all look more correct than before). 

The only other change were couple type-comparison (E721) fixes, these should be reviewed carefully.

I also turned off verbose pytest output in CI since it produces thousands of lines and is very difficult to navigate in Github UI.

----

### type-comparison (E721)

Derived from the **pycodestyle** linter.

#### What it does
Checks for object type comparisons using `==` and other comparison
operators.

#### Why is this bad?
Unlike a direct type comparison, `isinstance` will also check if an object
is an instance of a class or a subclass thereof.

If you want to check for an exact type match, use `is` or `is not`.

#### Known problems
When using libraries that override the `==` (`__eq__`) operator (such as NumPy,
Pandas, and SQLAlchemy), this rule may produce false positives, as converting
from `==` to `is` or `is not` will change the behavior of the code.

For example, the following operations are _not_ equivalent:
```python
import numpy as np

np.array([True, False]) == False
# array([False,  True])

np.array([True, False]) is False
# False
```

#### Example
```python
if type(obj) == type(1):
    pass

if type(obj) == int:
    pass
```

Use instead:
```python
if isinstance(obj, int):
    pass
```

### CODE REVIEW

File: .github/workflows/ci-code.yml
Comment (danielhollas) [inline .github/workflows/ci-code.yml:107]: Running pytest in verbose mode produces more than 2000 lines of output, which is extremely annoying in Github UI.

File: src/aiida/cmdline/groups/dynamic.py
Comment (danielhollas) [inline src/aiida/cmdline/groups/dynamic.py:180]: This is one of the few changes, I think this is correct, but please check.

File: src/aiida/engine/processes/workchains/workchain.py
Comment (danielhollas) [inline src/aiida/engine/processes/workchains/workchain.py:211]: Same here

File: src/aiida/cmdline/groups/dynamic.py
Comment (agoscinski) [reply src/aiida/cmdline/groups/dynamic.py:180]: The only way a different result can be achieved is when a custom type overwrites the equal function. That is not the case for `NoneType`, so it is ok.

File: src/aiida/engine/processes/workchains/workchain.py
Comment (agoscinski) [reply src/aiida/engine/processes/workchains/workchain.py:211]: `AttributeDict` is a class inheriting from `dict` so the `__eq__` function of the type is not overwritten so it is okay here

**Review by agoscinski — COMMENTED** [review]
I think you need to wait until #6630 is merged to fix the CI. Otherwise looks good to me

File: .pre-commit-config.yaml
Comment (danielhollas) [reply .pre-commit-config.yaml:49]: This was excluding files docs/source/topics/processes/include/snippets/functions/parse_docstring_expose_ipython.py and docs/source/topics/processes/include/snippets/functions/signature_plain_python_call_illegal.py from linting and formatting.

I've moved this configuration to pyproject.toml which I think is a much better place for it. For example, you can now run ruff also manually outside of pre-commit and it will have the same configuration.

**Comment by unkcpz** [comment]
@danielhollas I put this into "v2.7.0" in case there are some changes that we are not sure and will break the backward compatibility. Otherwise we can just have this for a patch release. What do you think?

**Comment by danielhollas** [comment]
There shouldn't be any breaking changes, but at the same time this certainly doesn't contain any bugfixes  so I don't see a reason why it would need to go in a patch release. v2.7.0 is fine

File: .pre-commit-config.yaml
Comment (agoscinski) [reply .pre-commit-config.yaml:49]: Ah this is reusing a yaml anchor! thanks!

**Review by agoscinski — APPROVED** [review]
I also don't see any reason for keeping it outside of a patch release if it simplifies the creation of it because of merge conflicts

File: .pre-commit-config.yaml
Comment (danielhollas) [reply .pre-commit-config.yaml:49]: What you said, yes :sweat_smile:

Comment (danielhollas) [reply .pre-commit-config.yaml:49]: FTR: I am refusing to google what is "yaml anchor" :-D

**Review by unkcpz — APPROVED** [review]
Looks all good!  Thanks @danielhollas

**Comment by danielhollas** [comment]
> I also don't see any reason for keeping it outside of a patch release if it simplifies the creation of it because of merge conflicts

True, there's a fair amount of churn in this PR due to import reorderings. In case you thought it important, I could split this into two commits, one with a simple

**Comment by unkcpz** [comment]
Thanks @danielhollas

**Pull request merged**
